File: ts/1408/1337.ts
function kWeakestRows(mat: number[][], k: number): number[] {
  const m = mat.length, n = mat[0].length
  const buckets: number[][] = new Array(m).fill(0).map(() => new Array(2).fill(0))
  const ans: number[] = []
  for (let i = 0; i < m; ++i) {
    let l = 0, r = n - 1, mid = 0
    while (l <= r) {
      mid = l + ((r - l) >> 1)
      if (mat[i][mid] === 1) l = mid + 1
      else r = mid - 1
    }
    buckets[i] = [i, l]
  }
  buckets.sort((a, b) => a[1] - b[1])

  for (let i = 0; i < k; ++i) {
    ans.push(buckets[i][0])
  }

  return ans
};

File: py3/1408/1337.py
from typing import List

class Solution:
    def kWeakestRows(self, mat: List[List[int]], k: int) -> List[int]:
        def getSecond(list):
            return list[1]
        m, n = len(mat), len(mat[0])
        buckets = []
        for i in range(m):
            l, r = 0, n - 1
            while l <= r:
                m = l + (r - l) // 2
                if mat[i][m] == 1:
                    l = m + 1
                else:
                    r = m - 1
            buckets.append([i, l])

        buckets.sort(key=getSecond)

        ans = []
        for i in range(k):
            ans.append(buckets[i][0])

        return ans